src/lib.rs

```diff
@@ -55,7 +56,7 @@ fn min_opt(left: u64, right: Option<u64>) -> u64 {
 }
 
 #[allow(dead_code)]
-#[cfg(unix)]
+#[cfg(all(unix, not(target_os="openbsd")))]
```

Comment:
I think based on what you said that you actually want to factor this into narrower private helpers and have the RLIMIT_AS one be stubbed for openbsd, but do still call RLIMIT_DATA on openbsd. The current patch just disables ulimited_memory detection entirely on openbsd.

It would also be great to have a workflow setup for openbsd; that might be as simple as changing the `skip_pr` field for opensbsd in the action-templates list.

Reply:
> The current patch just disables ulimited_memory detection entirely on openbsd.

That was the point of this patch, I just wanted a project that depends on this library to compile.
Unfortunately I don't know how your project works and I'm quite new to OpenBSD.
I haven't even figured out how to get Rust beta or nightly running on OpenBSD. I only have stable from ports.

> It would also be great to have a workflow setup for openbsd; that might be as simple as changing the `skip_pr` field for opensbsd in the action-templates list.

That's probably not gonna work since OpenBSD is tier 3 and doesn't have rustup support; and "[We do not even attempt tier three builds](https://github.com/riscygeek/effective-limits.rs/tree/master/ci/actions-templates#targets-we-need-to-build)".

Reply:
Ok. I can probably do that split sometime. For now perhaps a note that OpenBSD support is a stub - that more is possible but not done yet.